### ISSUE TITLE
incorrect diagnostic: `using a '.' after a '?.' might fail`

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#1192]
minimal example:

```js
undefined?.a.b
```

quick-lint reports: `using a '.' after a '?.' might fail, since '?.' might return 'undefined'.`, however this is not true.

Running the above code shows that optional chaining short-circuits the _entire property access expression_ to `undefined`, not just the first property access.

```
❯ deno
Deno 1.39.4
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> undefined?.a.b.c.d
undefined
```

idk if there's a "correct" version of this diagnostic, or if we should just delete the whole thing

**Comment by vegerot** [comment]
A better version of this diagnostic would be to check for non-null assertions, which are incorrect.

```ts
undefined?.a!.b
```

**Comment by strager** [comment]
Fixed in Git commit 434d08514d2ab0bc70b5ecb40f1e1e7c1256b6ae. I will ship the fix in version 3.2.0.

**Comment by strager** [comment]
Fixed released in [version 3.2.0](https://quick-lint-js.com/releases/#3.2.0).
